pkg/policymutation/policymutation.go

```diff
@@ -407,10 +409,17 @@ func CanAutoGen(policy *kyverno.ClusterPolicy, log logr.Logger) (applyAutoGen bo
 			return false, "none"
 		}
 
+		if needAutogen = hasAutogenKinds(match.Kinds); !needAutogen {
+			needAutogen = hasAutogenKinds(exclude.Kinds)
```

Comment:
This is odd. If I write a policy that does **not** apply to _Pods_ or, say, _Deployments_ by way of exclusion, why does that induce automatic generation of rules?

Reply:
Good point, we're doing the same [here](https://github.com/kyverno/kyverno/blob/9e10eef42203b92d5d3c94a30a378d89c46ab29b/pkg/policymutation/policymutation.go#L406-L408). @vyankyGH is it safe to remove the checks for `exclude.(Kinds/Any/All)`?